presto-main/src/main/java/com/facebook/presto/operator/scalar/sql/MapSqlFunctions.java

```diff
@@ -67,4 +66,15 @@ public static String mapTopNValuesComparator()
     {
         return "RETURN IF(n < 0, fail('n must be greater than or equal to 0'), slice(reverse(array_sort(remove_nulls(map_values(input)), f)) || filter(map_values(input), x -> x is null), 1, n))";
     }
+
+    @SqlInvokedScalarFunction(value = "map_top_n_keys_by_value", deterministic = true, calledOnNullInput = true)
+    @Description("Returns the keys of top N values of the given map in descending order as per the natural ordering.")
+    @TypeParameter("K")
+    @TypeParameter("V")
+    @SqlParameters({@SqlParameter(name = "input", type = "map(K, V)"), @SqlParameter(name = "n", type = "bigint")})
+    @SqlType("array<K>")
+    public static String mapTopNKeysByValue()
+    {
+        return "RETURN IF(n < 0, fail('n must be greater than or equal to 0'), transform(slice(array_sort_desc(transform(map_entries(input), x -> ROW(x[2], x[1]))), 1, n), x -> x[2]))";
```

Comment:
Use ARRAY_MAX_BY?

Reply:
not sure how is that gonna work. ARRAY_MAX_BY returns only the max value not max N values